Bubble refinement

PaulTashima opened the pull request.

buzzbuzz55 committed:
Bubble Refinement

• Attack probabilities updated
• Movement of bubble updated to use motion curve
• Movement of bubble updated to target player location at time of creation
• Bubble now removes itself on player contact.

buzzbuzz55 committed:
Bubble sprites

Added idle and death animations. Size should be adjusted - feels a bit small at the moment.

rhonig22 commented on Assets/Scripts/Enemy/Minions/BubbleController.cs:
```diff
@@ -0,0 +1,45 @@
+    {
+        _elapsedTime += Time.deltaTime;
+        _percentageComplete = _elapsedTime / _desiredDuration;
+        transform.position = Vector3.Lerp(_startPosition, _endPosition, _curve.Evaluate(_percentageComplete));
```
Maybe this logic should be in Move, rather than FixedUpdate, since FixedUpdate is calling Move already.

PaulTashima replied:
When i move this to Move the bubbles stop moving. Any idea why it's not equivalent?

rhonig22 commented on Assets/Scripts/Enemy/Minions/BubbleController.cs:
```diff
@@ -0,0 +1,45 @@
+        if (collision.collider != null && collision.collider.CompareTag("Player"))
+        {
+            _spriteAnimator.SetTrigger("death");
+            // This is a bad workaround - would love to find a way to trigger EnemyDeath after animation completion
```
Maybe tomorrow I can walk you through adding an animation behavior code behind to the death animation, so that you can trigger Destroy at the end of the animation?

buzzbuzz55 committed:
Update BubbleController.cs

Relocates the movement code from FixedUpdate to Move.
- Removes the FixedUpdate block.
